### PR TITLE
Chapter5

### PR DESCRIPTION
**Pull request by OleJune**

### CODE REVIEW

File: chapter5/5-1.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import time
+seconds = time.time()
+days = seconds//(60*60*24)
+local_time = time.strftime('%H:%M:%S')
```
Comment (auraz) [inline chapter5/5-1.py:4]: The idea of task is to get local time from the seconds (similar like you did with days). Please update the code. (w/o calling strftime)

File: chapter5/5-4.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+def recurse(n, s):
```
Comment (auraz) [inline chapter5/5-4.py:1]: please send stack diagram photo, as per task requirements.

File: chapter5/5-4.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+	"""
+	Prints the value of 's' when 'n == 0'.
```
Comment (auraz) [inline chapter5/5-4.py:3]: Please update docstring with explaining the idea of this function. E.g. Right now docstring is explaining only lines 9 and 10.

File: chapter5/5-2_prompt.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+def check_fermat(a, b, c, n):
```
Comment (auraz) [inline chapter5/5-2_prompt.py:1]: Good!

File: chapter5/5-3.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
+def is_triangle(a, b, c):
```
Comment (auraz) [inline chapter5/5-3.py:1]: Good!

File: chapter5/5-4.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+	Parameters:
+		n: positive int
+		s: int or float
+	"""
+	if n == 0:
+		print(s)
+	else:
+		recurse(n-1, n+s)
+recurse(3, 0)
+#recurse(-1, 0)  #this is an infinite recursion that causes a runtime error
```
Comment (auraz) [inline chapter5/5-4.py:14]: Good!

File: chapter5/5-6.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
+import turtle as t
```
Comment (auraz) [inline chapter5/5-6.py:1]: Good!

**Review by auraz — CHANGES_REQUESTED** [review]
Overall looks very good! Please address minor comments.

File: chapter5/5-1.py (hunk 1 above)
Comment (OleJune) [reply chapter5/5-1.py]: Fixed

File: chapter5/5-4.py (hunk 3 above)
Comment (OleJune) [reply chapter5/5-4.py]: Fixed

File: chapter5/5-4.py (hunk 2 above)
Comment (OleJune) [reply chapter5/5-4.py:1]: Sent

File: chapter5/5-4.py [hunk 8]
```diff
@@ -0,0 +1,18 @@
+def recurse(n, s):
+	"""
+	Calculates and prints the value of argument2 (s) after calling itself 
+	until it reaches the base case where argument1 (n) equals 0.
+
+	# or: The function calls itself until argument1 (n) equals 0
```
Comment (auraz) [inline chapter5/5-4.py:6]: This line is good

File: chapter5/5-4.py [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+	#		and then returns the value of argument2 (s).
```
Comment (auraz) [inline chapter5/5-4.py:7]: This line needs extension, in the end of the line please add, and in this case, value of s will be equal to?

**Review by auraz — CHANGES_REQUESTED** [review]
Another pass, very good, please continue.

**Review by auraz — APPROVED** [review]
Good job!
You can merge!

**Pull request merged**